src/map/MapTypes.jl

```diff
@@ -0,0 +1,42 @@
+###############################################################################
+#
+#   GeneralType
+#
+###############################################################################
+
+Map(::Type{T}) where T <: AbstractAlgebra.Map = supertype(T)
```

Comment:
You might just be able to set this equal to AbstractAlgebra.Map rather than redefine it. Not 100% sure that will work, given that you also have a type of that name, but might be worth a try, since then it will work for all maps, not just Singular ones.

Reply:
Changed it. Seems to work.